specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/storage.json

```diff
@@ -3215,6 +3215,11 @@
           "x-ms-client-name": "IsLocalUserEnabled",
           "description": "Enables local users feature, if set to true"
         },
+        "enableExtendedGroups": {
```

Comment:
>enableExtendedGroups [](http://example.com/codeflow?start=9&length=20)

Can you replace this boolean to an enum? Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way. Here for example there could be an enum for the state of the extended groups with values like Disabled,Enabled

Reply:
Based on offline discussion,

[12:23 PM] Razvan Badea
Hi, you can add your comments to the pr about it and so I can close it based on it. ARM recommends to use enum so that in place of adding new bool properties you could add a new enum value. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. 
In this case I saw you have more boolean properties but if they are related you can replace them, like isLocalUserEnabled and enableExtendedGroups sound like that could be converted for example to a userFeature with [None, LocalUser, LocalUserWithExtendedGroups], if it can be grouped like this.
 
 [1:49 PM] Nagendra Tomar
For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. 
Razvan Badea, above is a perfect example for using enums and using two booleans in this case is plain wrong.
In this specific case being discussed here, there's already a LocalUser object which has already well defined CRUD APIs which are GA'ed long back. This change from Chandan Agrawal is adding a new feature which enables a LocalUser to have support for "extended groups" (where the user can be part of many more groups). 
So this is still a LocalUser (so it won't be correct to call it LocalUserWithExtendedGroups) and hence adding a boolean property makes more sense. If you disagree, let's discuss over a call.
 
[4:08 PM] Razvan Badea
I see, ok for me, you should also add this comments on the pr as my on call shift ended and the new reviewer could see it and sign it off faster in case there are no other comments

Reply:
@razvanbadea-msft can I close this myself or new reviewer will do that?